packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -455,6 +505,115 @@ export class WalletFactory {
         return false;
     }
 
+    private isShadowSignerEnabled(chain: Chain, options?: WalletOptions): boolean {
+        return (chain === "solana" || chain === "stellar") && options?.shadowSigner?.enabled !== false;
+    }
+
+    private async addShadowSignerToDelegatedSignersIfNeeded<C extends Chain>(
```

Comment:
we're adding too much to the wallet factory and this is specific to ncs-signer. how about having this moved to ncs-signer and in wallet creation, if we're using ncs-signer, we call the getShadowSigner method and then add it to the delegated signers array?